src/tm-threads.c

```diff
@@ -2190,14 +2190,26 @@ void TmThreadsSetThreadTimestamp(const int id, const SCTime_t ts)
     int idx = id - 1;
     Thread *t = &thread_store.threads[idx];
     t->pktts = ts;
-    struct timeval systs;
-    gettimeofday(&systs, NULL);
-    t->sys_sec_stamp = (uint32_t)systs.tv_sec;
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+
+#ifdef DEBUG
+    SCTime_t now = SCTIME_FROM_TIMEVAL(&tv);
+    if (t->sys_sec_stamp.secs != 0) {
+        SCTime_t tmpts = SCTIME_ADD_SECS(t->sys_sec_stamp, 3);
+        if (SCTIME_CMP_LT(tmpts, now)) {
+            SCLogDebug("%s: thread slept for %u secs", t->name, (uint32_t)(now.secs - tmpts.secs));
+        }
+    }
+#endif
+
+    t->sys_sec_stamp = SCTIME_FROM_TIMEVAL(&tv);
     SCMutexUnlock(&thread_store_lock);
 }
 
 bool TmThreadsTimeSubsysIsReady(void)
 {
+    static SCTime_t nullts;
```

Comment:
`SCTIME_INITIALIZER` not needed in this case?

Reply:
I'll add it